### PR TITLE
8257162: Initialize ThreadLocalAllocBuffer members

### PR DESCRIPTION
**Pull request by tschatzl**
Hi all,

  can I have reviews for this change that adds initialization for all ThreadLocalAllocBuffer members so that when inspecting threads' TLABs the values are not random too?

Testing: tier1,tier2

Thanks,
  Thomas

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8257162](https://bugs.openjdk.java.net/browse/JDK-8257162): Initialize ThreadLocalAllocBuffer members

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**) ⚠️ Review applies to 9b7c212742f5cd1b291d57bd15d2f888bc1fca4e
 * [Albert Mingkun Yang](https://openjdk.java.net/census#ayang) (@albertnetymk - Author)
 * [Stefan Johansson](https://openjdk.java.net/census#sjohanss) (@kstefanj - **Reviewer**)
 * [Per Liden](https://openjdk.java.net/census#pliden) (@pliden - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/1458/head:pull/1458`
`$ git checkout pull/1458`

### CODE REVIEW

File: src/hotspot/share/gc/shared/threadLocalAllocBuffer.cpp [hunk 1]
```diff
@@ -37,6 +37,27 @@ size_t       ThreadLocalAllocBuffer::_max_size = 0;
...
+  _bytes_since_last_sample_point(0),
+  _number_of_refills(0),
+  _fast_refill_waste(0),
+  _slow_refill_waste(0),
+  _gc_waste(0),
+  _slow_allocations(0),
+  _allocated_size(0),
+  _allocation_fraction(TLABAllocationWeight) {
+
+  // do nothing.  tlabs must be inited by initialize() calls
```
Comment (shipilev) [inline src/hotspot/share/gc/shared/threadLocalAllocBuffer.cpp:58]: Wording: `Do nothing. TLABs must be initialized by initialize() calls.`

**Review by shipilev — APPROVED** [review]
Looks good to me.

**Review by albertnetymk — APPROVED** [review]
I believe these extra stores will not introduce much overhead.

File: src/hotspot/share/gc/shared/threadLocalAllocBuffer.cpp [hunk 2]
```diff
@@ -37,6 +37,27 @@ size_t       ThreadLocalAllocBuffer::_max_size = 0;
...
+  _number_of_refills(0),
+  _fast_refill_waste(0),
+  _slow_refill_waste(0),
+  _gc_waste(0),
+  _slow_allocations(0),
+  _allocated_size(0),
+  _allocation_fraction(TLABAllocationWeight) {
+
+  // do nothing. TLABs must be inited by initialize() calls
+}
```
Comment (kstefanj) [inline src/hotspot/share/gc/shared/threadLocalAllocBuffer.cpp:59]: Any reason for the extra blank line here? I would also use a capital D in "Do nothing".

**Review by kstefanj — APPROVED** [review]
Looks good, just this comment on the comment that you can skip if you don't agree.

**Review by pliden — APPROVED** [review]

**Comment by tschatzl** [comment]
Thanks @pliden @kstefanj @albertnetymk @shipilev for your reviews

/integrate

**Pull request closed without merging**